src/main.py

```diff
@@ -1,18 +1,30 @@
 from os import getenv
 
+from api.rate_limiter import RateLimiter
+from api.rootme_api import RootMeApi
+
 import discord
+import asyncio
 
 SECRET = getenv('USELESS_SECRET')
 MODE = getenv('MODE')
 
-def main():
+async def main():
```

Comment:
so you choose to use async everywhere? would we just **not** use it instead?

Reply:
well if you use asynchronous function somewhere you should `await` every function that is `async` and a function that awaits another one must be awaited as well. So as soon as you use it the main becomes `async`.

Reply:
ok I get it tks! I'll trust you, does it seem mandatory or could we avoid it entirely?

Reply:
As soon as you use `await` or `async` in one place, you must use it for all the **callers**.
So I don't think we can nor should avoid it.